### ISSUE TITLE
AbortError: The play() request was interrupted by a new load request.

### ISSUE THREAD

**Issue by andreysamode** [ryanheise/just_audio#1107]
**Which API doesn't behave as documented, and how does it misbehave?**
In Chrome on the web version, when switching to a new song while the first song is still loading, I get the following error that interrupts the debug process:
`AbortError: The play() request was interrupted by a new load request. https://goo.gl/LdLk22`

**Minimal reproduction project**
https://github.com/mrandrey/just_audio_bug_report/

**To Reproduce (i.e. user steps, not code)**
1. Launch in debug mode in Chrome
2. Quickly click on Song 1, Song 2, Song 3, Song 2, Song 1, etc.
3. The debug process should interrupt with the error.

**Error messages**

```
AbortError: The play() request was interrupted by a new load request. https://goo.gl/LdLk22
```

**Expected behavior**
I expect the loading/buffering to stop, then the new song to load.

**Desktop (please complete the following information):**
 - OS: Mac OS Sonoma
 - Browser: Chrome

**Smartphone (please complete the following information):**
 - The problem is only on the web version.

**Flutter SDK version**
```
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.13.6, on macOS 14.0 23A344 darwin-x64, locale en-US)
[✓] Android toolchain - develop for Android devices (Android SDK version 34.0.0)
[✓] Xcode - develop for iOS and macOS (Xcode 14.3.1)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2022.3)
[✓] Connected device (2 available)
[✓] Network resources

• No issues found!
```

**Additional context**
The project is to have an album with preview songs and an ability to go to the "full view" on another screen to listen to the entire song if the user has purchased it. If the user clicks on preview, then quickly clicks on the full song button, then they'll get an error because the preview is still buffering while the full song on the next screen is already loading the next song. I need to stop the buffering/loading and load the next song. Ideally, I'd like to load only what the user clicks on, so I'd rather not use a playlist as the solution.

Thanks for looking into this!

**Comment by CooolWolf** [comment]
I had this problem too

**Comment by CooolWolf** [comment]
I think the js implementation should be modified

`<video id="video"></video>
<button id="button"></button>

<script>
  button.addEventListener('click', onButtonClick);

  function onButtonClick() {
    // This will allow us to play video later...
    video.load();
    fetchVideoAndPlay();
  }

  function fetchVideoAndPlay() {
    fetch('https://example.com/file.mp4')
    .then(response => response.blob())
    .then(blob => {
      video.srcObject = blob;
      return video.play();
    })
    .then(_ => {
      // Video playback started ;)
    })
    .catch(e => {
      // Video playback failed ;(
    })
  }
</script>`

**Comment by linchen2chris** [comment]
same here

**Comment by max-anders** [comment]
same, in fact, the behavior of the play() future is strange to me, I remember I had another problem with the fact that it didn't complete

**Comment by BrockMekonnen** [comment]
I had the same problem, too

**Comment by ryanheise** [comment]
Does anyone have a repeatable way to reproduce this? Following the reproduction steps, I have managed to reproduce it only once out of many attempts, which means I have no way to test whether a fix actually worked.

As for the fix, I did fix this type of error in the past, but apparently only when `play` is interrupted by `pause`. I can extend this same approach to work for `load`, but will need a way to test.

**Comment by ryanheise** [comment]
I have created a potential fix in PR #1460 

I am going blind on this one (without a repeatable way to reproduce this), so I will rely on you to test it and verify that it resolves your issue. Therefore, please let me know if it is resolved, and then I can release the fix.

Note that because just_audio has a dependency on just_audio_web from pub, you will need to override this. I've left instructions in the comments of the pubspec.yaml files in this repo. So basically, clone this repo, specifically this branch, and comment in/out the lines in the pubspec.yaml to point to this fixed version of just_audio_web.

**Comment by vongrad** [comment]
There seems to be the same issue on Android

**Comment by ryanheise** [comment]
@vongrad , what you're probably thinking of is the `PlayerInterruptedException` which isn't a bug, but the documented behaviour of just_audio if you try to interrupt the player while it's in the process of already loading something. You can either avoid interrupting a load in progress, or you can do it and be prepared to catch and handle the case where a load was interrupted.

**Comment by ryanheise** [comment]
On the present issue, I've now published the above fix, with the caveat that I haven't been able to test it. I've been out of action for the past few days due to a medical situation, but it looks like after returning, nobody else has been able to test it either. In this situation, I'm putting it out there to get some real world testing. If anyone encounters this error again, they will report it as a new issue, and I can address it then.
